pkg/api/validation/validation.go

```diff
@@ -2575,6 +2575,10 @@ func ValidateServiceUpdate(service, oldService *api.Service) field.ErrorList {
 		allErrs = append(allErrs, ValidateImmutableField(service.Spec.ClusterIP, oldService.Spec.ClusterIP, field.NewPath("spec", "clusterIP"))...)
 	}
 
+	if oldService.Spec.ClusterIP == api.ClusterIPNone {
+		allErrs = append(allErrs, ValidateImmutableField(service.Spec.ClusterIP, oldService.Spec.ClusterIP, field.NewPath("spec", "clusterIP"))...)
```

Comment:
This block is identical to above.  Can we just flatten this to a unilateral `ValidateImmutableField` call?  I can't think of any valid update of this field.

CREATE sets "" -> allocated
UPDATE A -> B is disallowed
UPDATE A -> "" is disallowed
UPDATE A -> "None" is disallowed
UPDATE None -> B is disallowed
UPDATE None -> "" is disallowed
UPDATE None -> "None" is disallowed

We should never every see it as ""